matchers/matchers.go

```diff
@@ -33,82 +33,93 @@ type matcherFactory struct {
 	newCommentsMatcher func(callback state.CommentCallback) CommentMatcher
 }
 
-var (
-	once            sync.Once
-	standardMatcher TodoMatcher
-	scriptsMatcher  TodoMatcher
-	phpMatcher      TodoMatcher
-	pythonMatcher   TodoMatcher
-	groovyMatcher   TodoMatcher
-)
-
 var (
 	standardMatcherFactory = &matcherFactory{
-		func(customTodos []string) TodoMatcher {
-			once.Do(func() {
-				if standardMatcher == nil {
-					standardMatcher = standard.NewTodoMatcher(customTodos)
-				}
-			})
-			return standardMatcher
-		},
+		func() func([]string) TodoMatcher {
+			var once sync.Once
+			var matcher TodoMatcher
+			return func(customTodos []string) TodoMatcher {
+				once.Do(func() {
+					matcher = standard.NewTodoMatcher(customTodos)
+				})
+				return matcher
+			}
+		}(),
 		func(callback state.CommentCallback) CommentMatcher {
 			return standard.NewCommentMatcher(callback, false)
 		},
 	}
 	standardMatcherWithNestedMultilineCommentsFactory = &matcherFactory{
-		func(customTodos []string) TodoMatcher {
-			once.Do(func() {
-				if standardMatcher == nil {
-					standardMatcher = standard.NewTodoMatcher(customTodos)
-				}
-			})
-			return standardMatcher
-		},
+		func() func([]string) TodoMatcher {
+			var once sync.Once
+			var matcher TodoMatcher
+			return func(customTodos []string) TodoMatcher {
+				once.Do(func() {
+					matcher = standard.NewTodoMatcher(customTodos)
+				})
+				return matcher
+			}
+		}(),
 		func(callback state.CommentCallback) CommentMatcher {
 			return standard.NewCommentMatcher(callback, true)
 		},
 	}
 	scriptsMatcherFactory = &matcherFactory{
-		func(customTodos []string) TodoMatcher {
-			once.Do(func() {
-				scriptsMatcher = scripts.NewTodoMatcher(customTodos)
-			})
-			return scriptsMatcher
-		},
+		func() func([]string) TodoMatcher {
+			return func(customTodos []string) TodoMatcher {
+				var once sync.Once
+				var matcher TodoMatcher
```

Comment:
Why are these inside the returned function instead of being in the closure, similar to how you've done it for `standard` matchers?

Reply:
Apparently, I was distracted. I corrected this misprint.